docs/src/components/NoSSR.tsx

```diff
@@ -0,0 +1,18 @@
+'use client';
+import * as React from 'react';
+
+export function NoSSR(props: { defer?: boolean; children: any }): React.JSX.Element {
```

Comment:
In the past I've sometimes used something like

```tsx
import * as React from 'react';

const subscribe = () => () => {}
const getSnapshot = () => false
const getServerSnapshot = () => true

export function NoSsr({ children }) {
  // true during SSR and hydration
  const isSsr = React.useSyncExternalStore(subscribe, getSnapshot, getServerSnapshot);
  return isSsr ? null : children;
}
```

This avoids the extra renders caused by `useLayoutEffect`. Haven't benchmarked it though.

Reply:
Don't care too much about perf, this is just to avoid SSR so I can benchmark the component's rendering client-side.

Reply:
It's a new component introduced in the codebase. 6 months from now, maybe another maintainer may want to use this not realizing this is only to be used under the `DISABLE_DEMO_SSR` feature flag.